doc/decisions/frontend_end_to_end_testing.md

```diff
@@ -0,0 +1,105 @@
+# E2E Testing
+
+## Problem
+
+There is currently no existing e2e tests locally and in the pipeline.
+This allows for increased frontend bugs and regression.
+It lacks profound decision making which framework to pick.
+
+## Constraints
+
+1. [Behavior-driven development](https://www.selenium.dev/documentation/test_practices/testing_types/#behavior-driven-development-bdd)
+   - a) We want to have .feature files and stick to the Gherkin syntax.
+2. [Page Object Design Pattern](https://www.selenium.dev/documentation/test_practices/encouraged/page_object_models/)
+   - a) A programming language that supports classes
+3. The library must be free software.
+4. Should be in Python due to simplicity and personal preference.
+5. The framework must have an active community.
+6. Simple is better than complex.
+7. Headless browser support so we can run the tests in our automated pipeline.
+
+## Assumptions
+
+1. Understanding of BDD practices and principles, including writing feature files using the Gherkin syntax.
+2. Familiarity with the Page Object Design Pattern and its implementation for creating modular and maintainable test code.
+3. Knowledge of end-to-end testing concepts and best practices.
+4. Basic understanding of Python programming language and its syntax.
+5. Having heard of pytest and its usage for test execution and assertions.
+
+## Solutions
+
+### Alternative [Java](https://www.java.com/de/) + [Selenium](https://www.selenium.dev)
+
+Java with Selenium is a popular and well-established choice for web test automation.
+While it remains a reliable option, there are personal preferences and concerns about Oracle's licensing changes, and there are other languages that can do the same with less boilerplate.
+
+### Alternative [Javascript](https://www.javascript.com) + [Playwright](https://playwright.dev)
+
+JavaScript with Playwright is a powerful combination for web test automation.
+However, concerns regarding potential code complexity and the temptation to write complicated workarounds influenced the decision to avoid JavaScript as the testing language.
```

Comment:
Hey guys,

leaving a small comment with my thoughts regarding the language here.

Playwright and Python is a great combination, which I’ve used in most of my previous projects. There is a reason why I chose JS + Playwright for PermaplanT for the scrapers, because of the tech stack. Since most of the team knew JS, I tried to avoid introducing an extra language to the stack. This could take extra time for them to switch between languages. 

However, there is no difference in the functionality that I’ve experienced myself.
The situation and the circumstances may have changed after a couple of months, of course, so the final decision is up to you! Hopefully, this helps you with the choice!